src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -172,6 +179,13 @@ def get_attrs(
             if item.owner_id is not None:
                 owners_by_type[actor_type_to_string(item.owner.type)].append(item.owner_id)
 
+            # Add the latest 10 activations to the results list
+            activations = AlertRuleActivations.objects.filter(alert_rule=item).order_by(
```

Comment:
I'm a little torn on how to best achieve this.

Here - we're querying AlertRuleActivations for _each_ alert rule 🤔 
This is because we _only_ want to fetch teh latest 10 activations per rule.

whereas other models, we're doing something like
```
triggers = AlertRuleTrigger.objects.filter(alert_rule__in=item_list)
```
So we're only doing a single fetch for all items associated with the alert rules we're fetching

🤔 is there a way for us to do both of these filters in a single query?

Reply:
@wedamija would you have an opinion here?

Reply:
the others are doing it with the `alert_rule__in` to avoid an `n+1` query. i'd recommend pulling this out using a list to filter and iterating over the results.

lemme lookup the relationship between these models, curious if i can whip up a query for ya.